### PR TITLE
[1.13.x] KOGITO-7217 Tests: Use starters for spring boot archetype

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-7217

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [ ] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [ ] Pull Request contains a link to the JIRA issue
- [ ] Pull Request contains a description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've ran `make before-pr` and everything is working accordingly
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run operator BDD testing</b>
  Please add comment: <b>/jenkins test</b>

* <b>Run RHPAM operator BDD testing</b>
  Please add comment: <b>/jenkins RHPAM test</b>
</details>

### CODE REVIEW

File: test/pkg/steps/deploy_files.go [hunk 1]
```diff
@@ -18,33 +18,48 @@ import (
...
+	ctx.Step(`^Deploy (quarkus|springboot) file "([^"]*)" from example service "([^"]*)" with env:$`, data.deployFileFromExampleServiceWithEnv)
 	ctx.Step(`^Deploy (quarkus|springboot) folder from example service "([^"]*)"$`, data.deployFolderFromExampleService)
 }
 
 // Deploy steps
 
 func (data *Data) deployFileFromExampleService(runtimeType, file, serviceName string) error {
 	sourceFilePath := fmt.Sprintf(`%s/%s/%s/%s`, data.KogitoExamplesLocation, serviceName, sourceLocation, file)
-	return deploySourceFilesFromPath(data.Namespace, runtimeType, serviceName, sourceFilePath)
+	return deploySourceFilesFromPathWithEnv(data.Namespace, runtimeType, serviceName, sourceFilePath, nil)
```
Comment (MarianMacik) [inline test/pkg/steps/deploy_files.go:43]: You could also call the new function (`deployFileFromExampleServiceWithEnv`) from here with empty table and reusing the formatting code. Up to you.

**Review by MarianMacik — APPROVED** [review]
Look good to me. Just a hind/idea, but not a big deal.

**Review by sutaakar — APPROVED** [review]

File: test/pkg/steps/deploy_files.go (hunk 1 above)
Comment (radtriste) [reply test/pkg/steps/deploy_files.go:43]: I would say that it does not change much for now. Let's see in case we would need to reuse the env part somehow

**Pull request merged**